packages/dd-trace/src/aiguard/sdk.js

```diff
@@ -0,0 +1,175 @@
+'use strict'
+
+const NoopAIGuard = require('./noop')
+const executeRequest = require('./client')
+const {
+  AI_GUARD_RESOURCE,
+  AI_GUARD_TARGET_TAG_KEY,
+  AI_GUARD_REASON_TAG_KEY,
+  AI_GUARD_ACTION_TAG_KEY,
+  AI_GUARD_BLOCKED_TAG_KEY,
+  AI_GUARD_META_STRUCT_KEY,
+  AI_GUARD_TOOL_NAME_TAG_KEY
+} = require('./tags')
+const log = require('../log')
+const { URL } = require('url')
+
+const ALLOW = 'ALLOW'
+
+class AIGuardAbortError extends Error {
+  constructor (reason) {
+    super(reason)
+    this.name = 'AIGuardAbortError'
+    this.reason = reason
+  }
+}
+
+class AIGuardClientError extends Error {
+  constructor (message, opts = {}) {
+    super(message)
+    this.name = 'AIGuardClientError'
+    if (opts.errors) {
+      this.errors = opts.errors
+    }
+    if (opts.cause) {
+      this.cause = opts.cause
+    }
+  }
+}
+
+class AIGuard extends NoopAIGuard {
+  constructor (tracer, config) {
+    super(tracer)
+
+    if (config.protocolVersion !== '0.4') {
```

Comment:
Is there a config switch to set the protocol? Might be good to mention in the error message if it's user-facing.

Reply:
Also I wonder whether we want to crash here, or just log an error like "AI Guard requires trace protocol version 0.4 (this can be set with DD...),observability of AI Guard results will not be available."
but not throw?